src/main/kotlin/calculate/Calculate.kt

```diff
@@ -0,0 +1,57 @@
+package calculate
+
+import java.util.ArrayDeque
+import java.util.Deque
+import java.util.stream.Collectors
+
+class Calculate(private val validations: List<CalculateValidation>) {
+
+    constructor() : this(listOf(EmptyCheckValidation(), OperatorValidation()))
+
+    fun calculate(input: String): Int {
+        validateInput(input)
+        val operators = input.split("\\d+".toRegex()).filter { it.isNotBlank() }
+        val numbers: Deque<Int> = extractAndConvertNumber(input, operators)
+        for (operator in operators) {
+            val operatedValue = operatate(numbers, operator)
+            numbers.addFirst(operatedValue)
+        }
+        return numbers.pop()
+    }
+
+    private fun validateInput(input: String) {
+        validations.forEach {
+            it.validate(input)
+        }
+    }
+
+    private fun extractAndConvertNumber(input: String, operators: List<String>): Deque<Int> {
+        var result = input.split(*operators.toTypedArray())
+        return result.stream()
+            .map(String::trim)
+            .map(String::toInt)
+            .collect(Collectors.toCollection { ArrayDeque() })
```

Comment:
java의 stream 기능 대신에 Kotlin Collection의 확장 함수들을 활용해보시면 좋을 것 같아요. 😊
```suggestion
        return result.map { it.trim() }
            .map { it.toInt() }
            .toCollection(ArrayDeque())
```

Reply:
오오 stream을 안쓰고 이런 기능이..! 감사합니다